### PR TITLE
 feat: add task to upload asset to Sauce for spec test job

### PR DESCRIPTION
**Pull request by tianfeng92**
## Description

Add a Cypress task `sauce:uploadAsset` (or a different name if preferred) to upload an asset to the spec test job.

Job example, I attached two pictures and a `test.log`: https://app.saucelabs.com/tests/b1ac7308d816424f91f3e973496ec0bc

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -134,6 +134,33 @@ Jobs reported to Sauce Labs:
...
+| `asset`          | `object` | Object containing information about the asset (file) to be uploaded to Sauce Labs.                                                              |
+| `asset.filename` | `string` | **Required.** The name of the file to upload, as it should appear in Sauce Labs (e.g., `"this-is-fine.png"`).                                   |
+| `asset.path`     | `string` | **Optional.** Path to the file on the local filesystem (e.g., `"pics/this-is-fine.png"`). Either `asset.path` or `asset.data` must be provided. |
+| `asset.data`     | `stream` | **Optional.** File data as a stream, used when directly providing file content. Either `asset.path` or `asset.data` must be provided.           |
+
+### Example Usage
+
+```javascript
+it("upload assets", () => {
+  cy.task("sauce:uploadAsset", {
```
[the remark this answers is not in the document]
Comment (alexplischke) [reply README.md:153]: Good question! I don't see _why not_. Especially as a variadic function.

[the remark this answers is not in the document]
Comment (tianfeng92) [reply README.md]: 👍

File: README.md [hunk 2]
```diff
@@ -134,6 +134,39 @@ Jobs reported to Sauce Labs:
...
+| `assets[].filename` | `string` | **Required.** The name of the file to upload, as it should appear in Sauce Labs (e.g., `"this-is-fine.png"`).                       |
+| `assets[].path`     | `string` | **Optional.** Path to the file on the local filesystem (e.g., `"pics/this-is-fine.png"`). Either `path` or `data` must be provided. |
+| `assets[].data`     | `stream` | **Optional.** File data as a stream, used when directly providing file content. Either `path` or `data` must be provided.           |
+
+### Example Usage
+
+```javascript
+it("upload assets", () => {
+  // Single file upload.
+  cy.task("sauce:uploadAssets", {
```
Comment (alexplischke) [inline README.md:154]: We have an integration test in CI. Would be good if it upload some file there as well 🙏

Comment (tianfeng92) [reply README.md:154]: 👍 Good idea!

File: src/index.ts [hunk 3]
```diff
@@ -20,6 +23,9 @@ export interface Options {
 let reporterInstance: Reporter;
 const reportedSpecs: { name: string; jobURL: string }[] = [];
 
+// Maps spec names to their associated assets, cached for later upload.
+let specAssetsMap: Map<string, Asset[]>;
```
[the remark this answers is not in the document]
Comment (tianfeng92) [reply src/index.ts:27]: `specToAssets` looks good to me. I'll use it.

**Review by alexplischke — APPROVED** [review]

**Pull request merged**